streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -152,9 +152,9 @@ public StreamsMetadata getLocalMetadata() {
         }
 
         if (globalStores.contains(storeName)) {
-            // global stores are on every node. if we dont' have the host info
+            // global stores are on every node. if we don't have the host info
             // for this host then just pick the first metadata
-            if (thisHost == UNKNOWN_HOST) {
+            if (thisHost.equals(UNKNOWN_HOST)) {
```

Comment:
Should we fix this in older branches (2.5/2.4), too? (ie follow up PR)

Reply:
Will do

Reply:
https://github.com/apache/kafka/pull/8919